app/controllers/timelogs_controller.rb

```diff
@@ -0,0 +1,54 @@
+class TimelogsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @timelogs = current_user.timelogs
+  end
+
+  def create
+    @timelog = current_user.timelogs.build(timelogs_params)
+    authorize @timelog
+    if @timelog.save
+      flash[:success] = "Timelog created"
+      redirect_to user_timelogs_path
+    else
+      render :new
+    end
+  end
+
+  def update
+    @timelog = Timelog.find(params[:format])
+    authorize @timelog
```

Comment:
we have similar duplications here, which can be separated into a separate method.